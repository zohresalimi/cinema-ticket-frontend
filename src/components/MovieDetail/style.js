import styled from "styled-components";

export const Wrapper = styled.div`
  font-family: ${(props) => props.theme.fontFamily.sf};
  display: flex;
  align-items: center;
  width: 100%;
  flex-direction: column;
  background-color: ${(props) => props.theme.colors.black};
`;

export const FullImage = styled.div`
  height: ${(props) => props.theme.fullImage.height};
  width: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;

  @media (max-width: 375px) {
    max-height: 540px;
  }
`;

export const Image = styled.img`
  width: 100%;
  max-width: 100%;
  max-height: 100%;
  object-fit: cover;
  object-position: bottom;
  height: 100%;
`;

export const ShadowBg = styled.div`
  background: linear-gradient(
    -180deg,
    transparent 50%,
    rgba(0, 0, 0, 0.68) 85%,
    #000
  );
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

export const Info = styled.div`
  position: relative;
  margin-top: -${(props) => props.theme.fullImage.height.replace("px", "") / 2.5}px;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  .movie-poster {
    width: 12rem;
    img {
      max-width: 100%;
      display: block;
      border-radius: 0.25rem;
    }
  }
  .movie-detail {
    text-transform: capitalize;
    margin-left: 1.5rem;
    h2 {
    }
    p {
      display: inline-block;
      margin: 5px 0;
    }
    .btn-wrapper {
      margin-top: 10px;
    }
    .movie-duration {
      text-transform: lowercase;
      margin: 0;
      span {
        margin: 0 5px;
      }
    }
  }
`;
export const MovieTitle = styled.h2`
  font-family: ${(props) => props.theme.fontFamily.sf};
  color: ${(props) => props.theme.colors.white};
  margin: 20px auto;
`;
