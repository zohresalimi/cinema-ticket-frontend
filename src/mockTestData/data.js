import React from "react";
import { Router } from "@reach/router";
import AppContext from "../store/context";
import Theme from "../Styles/Theme";

export const getTestStore = () => {
  return {
    dispatch: () => {},
    state: {
      user: {
        name: "",
        email: "",
      },
      movies: {
        children: [
          {
            _id: "5fa71823adc463397a50c501",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "trolls 2: world tour",
            originalTitle: "swedish",
            premiere: "2020-10-03T15:18:52.722Z",
            rooms: ["5fa653d0a7f1201999dd1ac0"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        premiered: [
          {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        upcoming: [
          {
            _id: "5fa71823adc463397a50c503",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "honest thief",
            originalTitle: "swedish",
            premiere: "2021-01-03T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        allMovies: {
          "5fa71823adc463397a50c501": "children",
          "5fa71823adc463397a50c502": "premiered",
          "5fa71823adc463397a50c503": "upcoming",
        },
      },
      cinemas: [
        {
          _id: "5fa64fe234fbbf9945f3a3ce",
          name: "filmstaden kista",
          purchaseEndTime: "21:30",
          purchaseStartTime: "8:00",
          rooms: ["5fa653c0643c06586bba7d78", "5fa653d0a7f1201999dd1ac0"],
        },
      ],
      ticket: {
        movie: {},
        status: false,
        quantity: 0,
        price: 0,
        seatNumbers: new Map(),
        user: {},
        showing: {},
      },
      showings: [
        {
          _id: "5fbe9bd31f58462b5c759b5c",
          capacity: 4,
          cinema: "5fa64fe234fbbf9945f3a3ce",
          endTime: null,
          movie: {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
          price: 100,
          room: { _id: "5fa653c0643c06586bba7d78" },

          seats: [
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
          ],
          startTime: "2020-11-13T16:00:00.000Z",
        },
      ],
    },
  };
};

export const WithProvider = (props) => {
  const { state, dispatch } = getTestStore();
  return (
    <Theme>
      <AppContext.Provider
        value={{ state: props.defaultValue || state, dispatch }}
      >
        <Router>{props.children}</Router>
      </AppContext.Provider>
    </Theme>
  );
};
