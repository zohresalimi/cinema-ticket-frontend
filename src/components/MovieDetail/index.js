import React, { useState } from "react";
import ReactPlayer from "react-player/youtube";
import { useTranslation } from "react-i18next";

import { Wrapper, FullImage, Image, ShadowBg, Info, MovieTitle } from "./style";
import PlayButton from "../PlayButton";
import Modal from "../Modal";
import { Container, Row, Button } from "../../Styles/StyleComponents";

function FeaturedMovie({ movie }) {
  const [playVideo, setPlayVideo] = useState(false);
  const [t] = useTranslation();

  return (
    <Wrapper>
      <FullImage>
        <Image alt="" src={movie.largeImage} />
        <ShadowBg />
        {playVideo && (
          <Modal setPlayVideo={setPlayVideo}>
            <div className="player-wrapper">
              <ReactPlayer
                className="react-player"
                url={movie.trailerUrl}
                config={{
                  youtube: {
                    playerVars: { modestbranding: 1, showinfo: 0, controls: 0 },
                  },
                }}
              />
            </div>
          </Modal>
        )}
        <PlayButton setPlayVideo={setPlayVideo} />
      </FullImage>
      <Container>
        <Row>
          <Info>
            <div className="movie-poster">
              <img src={movie.coverImage} alt="" />
            </div>
            <div className="movie-detail">
              <MovieTitle>{movie.name}</MovieTitle>
              <p>{t(movie.genre)}</p>
              <div className="movie-duration">
                <p>
                  {movie.duration.split(":")[0]}
                  <span>{t("hour")}</span>
                  {movie.duration.split(":")[1]}
                  <span>{t("min")}</span>
                </p>
                <span>|</span>
                <p>
                  {movie.age}
                  {t("year")}
                </p>
              </div>

              <div className="btn-wrapper">
                <Button>{t("buy ticket")}</Button>
              </div>
            </div>
          </Info>
        </Row>
      </Container>
    </Wrapper>
  );
}

export default FeaturedMovie;
