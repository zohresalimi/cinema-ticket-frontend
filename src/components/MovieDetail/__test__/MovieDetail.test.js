import * as React from "react";
import { render, act, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import { getTestStore, WithProvider } from "../../../mockTestData/data";
import MovieDetail from "../";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key) => key,
    i18n: { changeLanguage: jest.fn(), language: "sv" },
  }),
}));

function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider>
        <MovieDetail {...props} path="/" />
      </WithProvider>
    );
  });

  return component;
}

describe("MovieDetail Component Testing", () => {
  afterAll(() => {
    jest.resetAllMocks();
  });

  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("page should be scrolled", async () => {
    const movie = getTestStore().state.movies.premiered[0];
    const mockOnBookingClick = jest.fn(() => true);

    const { container, getByTestId } = await renderWrapper({
      onBookingClick: mockOnBookingClick,
      movie,
    });
    fireEvent.click(getByTestId("booking-scroll"));
    expect(mockOnBookingClick).toHaveBeenCalled();
    expect(container.firstChild).toMatchSnapshot();
  });
  test("should play video", async () => {
    const setState = jest.fn();
    jest.spyOn(React, "useState").mockImplementation(() => [true, setState]);
    const movie = getTestStore().state.movies.premiered[0];

    const { container } = await renderWrapper({
      movie,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
