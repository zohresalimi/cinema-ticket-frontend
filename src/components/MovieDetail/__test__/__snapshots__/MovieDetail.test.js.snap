// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail Component Testing page should be scrolled 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="style__Wrapper-ib79g7-0 jhlonx"
    data-testid="wrapper"
  >
    <div
      class="style__FullImage-ib79g7-1 hfTzCG"
    >
      <img
        alt=""
        class="style__Image-ib79g7-2 dpvpLG"
        src="https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp"
      />
      <div
        class="style__ShadowBg-ib79g7-3 cZHGqf"
      />
      <div
        class="style__Div-y1of7i-0 kOhgOG"
        data-testid="play-btn"
      >
        <div
          class="style__RoundBut-y1of7i-2 iqNHdy"
        >
          <a
            class="style__PlayBtn-y1of7i-1 BDDhv"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-play fa-w-14 "
              data-icon="play"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
    >
      <div
        class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
      >
        <div
          class="style__Info-ib79g7-4 hyXzhv"
        >
          <div
            class="movie-poster"
          >
            <img
              alt=""
              src="https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp"
            />
          </div>
          <div
            class="movie-detail"
          >
            <h2
              class="style__MovieTitle-ib79g7-5 lbHZIs"
            >
              follow me
            </h2>
            <p>
              family, children
            </p>
            <div
              class="movie-duration"
            >
              <p>
                1
                <span>
                  hour
                </span>
                31
                <span>
                  min
                </span>
              </p>
              <span>
                |
              </span>
              <p>
                7
                year
              </p>
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                class="StyleComponents__Button-sc-11erdxt-3 kZtNg"
                data-testid="booking-scroll"
              >
                buy ticket
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
      >
        <div>
          <p
            class="description"
          >
            Poppy and Kvist embark on an adventure that will take them far beyond
          </p>
          <p
            class="title"
          >
            director
          </p>
          <p>
            aaron woodley
          </p>
          <p
            class="title"
          >
            premiere
          </p>
          <p>
            <time
              datetime="1601911132722"
            >
              mån 5 okt
            </time>
          </p>
          <p
            class="title"
          >
            actor
          </p>
          <div
            class="displayInline"
          >
            <p>
              Holland Roden
            </p>
            <p>
              Ronen Rubinstein
            </p>
            <p>
              Keegan Allen
            </p>
          </div>
          <p
            class="title"
          >
            original language
          </p>
          <p>
            swedish
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieDetail Component Testing should play video 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="style__Wrapper-ib79g7-0 jhlonx"
    data-testid="wrapper"
  >
    <div
      class="style__FullImage-ib79g7-1 hfTzCG"
    >
      <img
        alt=""
        class="style__Image-ib79g7-2 dpvpLG"
        src="https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp"
      />
      <div
        class="style__ShadowBg-ib79g7-3 cZHGqf"
      />
      <div
        class="full-page"
      >
        <div
          class="video-wrapper"
        >
          <div
            class="close-btn"
            data-testid="close-btn"
            role="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times-circle fa-w-16 "
              data-icon="times-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="player-wrapper"
          >
            <div
              class="react-player"
              style="width: 100%; height: 100%;"
            >
              <div
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="style__Div-y1of7i-0 kOhgOG"
        data-testid="play-btn"
      >
        <div
          class="style__RoundBut-y1of7i-2 iqNHdy"
        >
          <a
            class="style__PlayBtn-y1of7i-1 BDDhv"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-play fa-w-14 "
              data-icon="play"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
    >
      <div
        class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
      >
        <div
          class="style__Info-ib79g7-4 hyXzhv"
        >
          <div
            class="movie-poster"
          >
            <img
              alt=""
              src="https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp"
            />
          </div>
          <div
            class="movie-detail"
          >
            <h2
              class="style__MovieTitle-ib79g7-5 lbHZIs"
            >
              follow me
            </h2>
            <p>
              family, children
            </p>
            <div
              class="movie-duration"
            >
              <p>
                1
                <span>
                  hour
                </span>
                31
                <span>
                  min
                </span>
              </p>
              <span>
                |
              </span>
              <p>
                7
                year
              </p>
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                class="StyleComponents__Button-sc-11erdxt-3 kZtNg"
                data-testid="booking-scroll"
              >
                buy ticket
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
      >
        <div>
          <p
            class="description"
          >
            Poppy and Kvist embark on an adventure that will take them far beyond
          </p>
          <p
            class="title"
          >
            director
          </p>
          <p>
            aaron woodley
          </p>
          <p
            class="title"
          >
            premiere
          </p>
          <p>
            <time
              datetime="1601911132722"
            >
              mån 5 okt
            </time>
          </p>
          <p
            class="title"
          >
            actor
          </p>
          <div
            class="displayInline"
          >
            <p>
              Holland Roden
            </p>
            <p>
              Ronen Rubinstein
            </p>
            <p>
              Keegan Allen
            </p>
          </div>
          <p
            class="title"
          >
            original language
          </p>
          <p>
            swedish
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieDetail Component Testing take snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="style__Wrapper-ib79g7-0 jhlonx"
    data-testid="wrapper"
  >
    <div
      class="style__FullImage-ib79g7-1 hfTzCG"
    >
      <svg
        aria-labelledby="movie-detail-loader-aria"
        height="600"
        role="img"
        viewBox="0 0 1500 600"
        width="1500"
      >
        <title
          id="movie-detail-loader-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#movie-detail-loader-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#movie-detail-loader-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs
          role="presentation"
        >
          <clippath
            id="movie-detail-loader-diff"
          >
            <rect
              height="9"
              rx="3"
              ry="3"
              width="131"
              x="504"
              y="429"
            />
            <rect
              height="9"
              rx="3"
              ry="3"
              width="258"
              x="503"
              y="451"
            />
            <rect
              height="12"
              rx="3"
              ry="3"
              width="170"
              x="503"
              y="484"
            />
            <circle
              cx="685"
              cy="118"
              r="59"
            />
            <rect
              height="278"
              rx="0"
              ry="0"
              width="194"
              x="273"
              y="285"
            />
            <rect
              height="48"
              rx="0"
              ry="0"
              width="104"
              x="503"
              y="514"
            />
          </clippath>
          <lineargradient
            id="movie-detail-loader-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#262626"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#575757"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#262626"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>
</div>
`;
